source/val/validate_misc.cpp

```diff
@@ -35,6 +35,37 @@ spv_result_t ValidateUndef(ValidationState_t& _, const Instruction* inst) {
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateShaderClock(ValidationState_t& _, 
+                                 const Instruction* inst) {
+  bool is_int32 = false, is_const_int32 = false;
+  uint32_t value = 0;
+  const uint32_t execution_scope = inst->word(3);
+  std::tie(is_int32, is_const_int32, value) = _.EvalInt32IfConst(execution_scope);
+
+  // Vulkan specific rules
+  if (spvIsVulkanEnv(_.context()->target_env)) {
+    if (value != SpvScopeSubgroup && value != SpvScopeDevice) {
+      return _.diag(SPV_ERROR_INVALID_DATA, inst) << "ExecutionScope should be "
+                                                     "ScopeSubgroup or ScopeDevice";
+    }
+  }
+
+  // Result Type must be a 64 - bit unsigned integer type or
+  // a vector of two - components of 32 -
+  // bit unsigned integer type 
+  const uint32_t result_type = inst->type_id();
+  if (((!_.IsUnsignedIntVectorType(result_type) ||
```

Comment:
should also check that the components of the vector are 32 bits wide.

Reply:
```
if (!(_.IsUnsignedIntScalarType(result_type) && _.GetBitWidth(result_type) == 64) &&
    !(_.IsUnsignedIntVectorType(result_type) && _.GetDimension(result_type) == 2 && _.GetBitWidth(result_type) == 32)) {
  // error
}
```

Then we don't need the new method.